editors/code/src/toolchain.ts

```diff
@@ -181,12 +180,6 @@ function lookupInPath(exec: string): boolean {
     return candidates.some(isFile);
 }
 
-function isFile(suspectPath: string): boolean {
-    // It is not mentionned in docs, but `statSync()` throws an error when
-    // the path doesn't exist
-    try {
-        return fs.statSync(suspectPath).isFile();
-    } catch {
-        return false;
-    }
+async function isFile(path: string): Promise<boolean> {
+    return ((await vscode.workspace.fs.stat(vscode.Uri.file(path))).type & vscode.FileType.File) !== 0;
```

Comment:
What happens here if the file doesn't exist?

Reply:
If I had to guess... [it throws](https://github.com/microsoft/vscode/blob/60d2aeee4b1c3104bd55c99679546168df4f6001/src/vs/vscode.d.ts#L7141). (Note: that's for `FileSystemProvider`, not `FileSystem`, but I'm assuming the latter is a wrapper over the former?) This should definitely be wrapped in a try/catch block ASAP.

I'm not very proud of my implementation for this function, either; I wanted to maintain the original function signature (i.e. with a string path, not `Uri`) so that it could be reused in `lookupInPath()`.

Looking at it now, I think it would be best to separate `isFile()` into two functions: one that takes a string path, and one that takes a `Uri`. The former should be a simple wrapper over the latter, translating the string into a `Uri` with `Uri.file(path)`.

This PR has already been merged, so tomorrow (later today?) I can try and clean this up in a new one.